src/Pos/Wallet/Web/ClientTypes.hs

```diff
@@ -131,16 +131,23 @@ mkCTxId = CTxId . CHash
 txIdToCTxId :: TxId -> CTxId
 txIdToCTxId = mkCTxId . sformat hashHexF
 
+convertTxOutputs :: [TxOut] -> [(CAddress, CCoin)]
+convertTxOutputs = map (addressToCAddress . txOutAddress &&& mkCCoin . txOutValue)
+
 mkCTx
     :: Address            -- ^ An address for which transaction info is forming
     -> ChainDifficulty    -- ^ Current chain difficulty (to get confirmations)
     -> TxHistoryEntry     -- ^ Tx history entry
     -> CTxMeta            -- ^ Transaction metadata
     -> CTx
-mkCTx addr diff THEntry {..} meta = CTx {..}
+mkCTx addr diff THEntry {..} meta =
+    -- placeholder, this functionality isn't implemented yet
+    let ctFrom = [] in
```

Comment:
@Pastafarianist It would be much cleaner (without involving additional explorer stuff) to just extend `TxHistoryEntry` to contain lists of input and output addresses. While deriving `TxHistoryEntries` we have access to all this info.